db_test.go

```diff
@@ -156,13 +156,128 @@ func TestDB_ImportExport(t *testing.T) {
 			// We have to reset the embed function, but otherwise the DB objects
 			// should be deep equal.
 			c.embed = nil
-			if !reflect.DeepEqual(orig, new) {
-				t.Fatalf("expected DB %+v, got %+v", orig, new)
+			if !reflect.DeepEqual(orig, newDB) {
+				t.Fatalf("expected DB %+v, got %+v", orig, newDB)
 			}
 		})
 	}
 }
 
+func TestDB_ImportExportSpecificCollections(t *testing.T) {
+	r := rand.New(rand.NewSource(rand.Int63()))
+	randString := randomString(r, 10)
+	path := filepath.Join(os.TempDir(), randString)
+	filePath := path + ".gob"
+	defer os.RemoveAll(path)
+
+	// Values in the collection
+	name := "test"
+	name2 := "test2"
+	metadata := map[string]string{"foo": "bar"}
+	vectors := []float32{-0.40824828, 0.40824828, 0.81649655} // normalized version of `{-0.1, 0.1, 0.2}`
+	embeddingFunc := func(_ context.Context, _ string) ([]float32, error) {
+		return vectors, nil
+	}
+
+	// Create DB, can just be in-memory
+	orig := NewDB()
+
+	// Create collections
+	c, err := orig.CreateCollection(name, metadata, embeddingFunc)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+
+	c2, err := orig.CreateCollection(name2, metadata, embeddingFunc)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+
+	// Add documents
+	doc := Document{
+		ID:        name,
+		Metadata:  metadata,
+		Embedding: vectors,
+		Content:   "test",
+	}
+
+	doc2 := Document{
+		ID:        name2,
+		Metadata:  metadata,
+		Embedding: vectors,
+		Content:   "test2",
+	}
+
+	err = c.AddDocument(context.Background(), doc)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+
+	err = c2.AddDocument(context.Background(), doc2)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+
+	// Export
+	err = orig.ExportToFile(filePath, false, "", name2)
+	if err != nil {
+		t.Fatal("expected no error, got", err)
+	}
+
+	dir := filepath.Join(path, randomString(r, 10))
+	defer os.RemoveAll(dir)
+
+	newPDB, err := NewPersistentDB(dir, false)
```

Comment:
Comparing with the existing test, here it could also help to explain the difference:
```suggestion
	// Instead of importing to an in-memory DB we use a persistent one to cover the behavior of immediate persistent files being created for the imported data
	newPDB, err := NewPersistentDB(dir, false)
```
(Or maybe there's better phrasing)